src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -128,19 +174,41 @@ class OrderedRVD(
 
   override def unpersist(): OrderedRVD = this
 
+  // FIXME: don't allow to change type
   def constrainToOrderedPartitioner(
     ordType: OrderedRVDType,
     newPartitioner: OrderedRVDPartitioner
   ): OrderedRVD = {
-
-    require(ordType.rowType == typ.rowType)
-    require(ordType.kType isPrefixOf typ.kType)
+    // FIXME: require newPartitioner disjoint
```

Comment:
are these FIXMEs addressed in later PRs?

Reply:
No, I missed those. I'll see if I can resolve them easily, otherwise I'll move them to my personal todo list. There's no correctness issue here, only API design.